submissions/Serge044/Classic-Frogger-Game/app.js

```diff
@@ -0,0 +1,151 @@
+const Enemy = function (x, y) {
+  this.width = 75;
+  this.x = x;
+  this.y = y;
+  this.border = { left: -100, right: 505 };
+  this.speed = getRandomNum(200, 300);
+  this.sprite = "images/tank-icon.png";
+};
+
+Enemy.prototype.update = function (dt) {
+  if (this.x < this.border.right) {
+    this.x += dt * this.speed + Math.random();
+  } else {
+    this.speed = getRandomNum(200, 300);
+    this.x = this.border.left;
+  }
+};
+
+function getRandomNum(min, max) {
+  return Math.floor(Math.random() * (max - min + 1)) + min;
+}
+
+Enemy.prototype.render = function () {
+  ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+const Player = function () {
+  this.width = 75;
+  this.x = 202;
+  this.y = 400;
+  this.stepX = 101;
+  this.stepY = 84;
+  this.border = { top: -20, right: 404, bottom: 400, left: 0 };
+  this.sprite = "images/char-nafo.png";
+};
+
+Player.prototype.render = function () {
+  ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+Player.prototype.toStartPosition = function () {
+  this.x = 202;
+  this.y = 400;
+};
+
+Player.prototype.update = function () {
+  if (this.y === this.border.top) {
+    setTimeout(() => {
+      swal(
+        "Congratulations!",
+        "Crimea finally under the Ukrainian flag 🇺🇦",
+        "success",
+        {
+          button: "Слава Україні!",
+        }
+      );
+      this.toStartPosition();
+    }, 100);
+  }
+  this.checkEnemyCollision();
+};
+
+Player.prototype.checkEnemyCollision = function () {
+  allEnemies.forEach((el) => {
+    if (
+      el.y == this.y &&
+      this.x <= el.x + el.width &&
+      this.x + this.width >= el.x
+    ) {
+      this.toStartPosition();
+    }
+  });
+};
+
+Player.prototype.handleInput = function (event) {
+  if (event === "ArrowUp" && this.y !== this.border.top) {
+    this.y -= this.stepY;
+  } else if (event === "ArrowDown" && this.y !== this.border.bottom) {
+    this.y += this.stepY;
+  } else if (event === "ArrowLeft" && this.x !== this.border.left) {
+    this.x -= this.stepX;
+  } else if (event === "ArrowRight" && this.x !== this.border.right) {
+    this.x += this.stepX;
+  }
+};
+
+const player = new Player();
+
+const enemy0 = new Enemy(-101, 64);
+const enemy1 = new Enemy(-101, 148);
+const enemy2 = new Enemy(-101, 232);
```

Comment:
(1) Mystery numbers.
(2) If a peer developer is tasked to add 20 more enemies should they add `enemy3` ... `enemy22` - 20 lines of code? Please make this fragment DRY so that adding new enemies requires as few effort as possible. Also use power of arrays. Ordinal numbers in variable names is a clear sign that you need an indexed array instead of "indexed" variables.